themes/bootstrap3/js/common.js

```diff
@@ -56,6 +81,27 @@ var VuFind = (function VuFind() {
     });
   };
 
+  var initClickHandlers = function initClickHandlers() {
+    window.addEventListener(
+      'click',
+      function handleClick(event) {
+        let elem = event.target;
+        if (elem.hasAttribute('data-click-callback')) {
+          return evalCallback(elem.dataset.clickCallback, event, {});
+        }
+      }
+    );
+    window.addEventListener(
+      'change',
+      function handleChange(event) {
+        let elem = event.target;
+        if (elem.hasAttribute('data-submit-on-change')) {
```

Comment:
I wonder if some of this generic refactoring would benefit from being a separate PR... and I also wonder if this data-submit-on-change mechanism could be used for some of our other existing functionality, like the language/theme drop-downs which are also being refactored by #2378. I feel like there are a lot of balls in the air with templates/themes, and it's getting increasingly difficult to decide how to move forward. Don't let me make this worse through this suggestion, but I'm wondering if there's an opportunity to unify a couple of open threads...

Reply:
Yes, the idea here is that these could be used for any other functionality as well. I'm open to extracting this to a separate PR or incorporating into another, but you're right in that there's so much mid-flight at the moment that it's difficult to decide on how to proceed.